### PR TITLE
CVE-2022-34917 (High) detected in kafka-clients-3.0.1.jar

### PR DESCRIPTION
**Pull request by vinayak15**
### Description
[Describe what this change achieves]
Update Kafka version to fix security vulnerability(#2095)

### Issues Resolved
[[List any issues this PR will resolve]
](https://github.com/opensearch-project/security/issues/2095)
Is this a backport? If so, please add backport PR # and/or commits #

### Testing
[Please provide details of testing done: unit testing, integration testing and manual testing]

### Check List
- [ ] New functionality includes testing
- [ ] New functionality has been documented
- [ ] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -98,7 +98,6 @@ forbiddenApisTest.enabled = false
 filepermissions.enabled = false
 forbiddenPatterns.enabled = false
 testingConventions.enabled = false
-// Conflicts between runtime kafka-clients:3.0.1 & testRuntime kafka-clients:3.0.1:test
```
Comment (peternied) [inline build.gradle:101]: This comment reference to the line below, can we activate jarhell with 3.0.2?

File: build.gradle [hunk 2]
```diff
@@ -307,7 +306,7 @@ dependencies {
     }
     implementation 'com.github.wnameless:json-flattener:0.5.0'
     implementation 'com.flipkart.zjsonpatch:zjsonpatch:0.4.4'
-    implementation 'org.apache.kafka:kafka-clients:3.0.1'
+    implementation 'org.apache.kafka:kafka-clients:3.0.2'
```
Comment (peternied) [inline build.gradle:309]: Can we make a variable for the kafka version so future changes don't need to update so many lines at once?

**Review by peternied — CHANGES_REQUESTED** [review]
Happy to merge after we get a resolution on the change around jarhell

File: build.gradle (hunk 1 above)
Comment (cwperks) [reply build.gradle:101]: @peternied The kafka change to resolve jar hell was merged into main and didn't make it into the 3.0.2 release.

Here's the change merged into main: https://github.com/apache/kafka/pull/12407

Looks like its still in the code of 3.0.2: https://github.com/apache/kafka/blob/25b1aea02e37da1457026f6ac82a66f81c878e45/build.gradle#L1261

Comment (peternied) [reply build.gradle:101]: Understood, but we should keep the comment until we've resolved the issue it mentions.  If the concern is about the version number being hardcoded to a different version, you could just delete them or replace them with `X.Y.Z`.

Comment (cwperks) [reply build.gradle:101]: Yes we should leave the comment in until the jar hell issue is resolved

**Review by DarshitChanpura — CHANGES_REQUESTED** [review]
Ty @vinayak15. Please rebase and sign-off the latest commit.

**Pull request closed without merging**